app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,52 +1,129 @@
-            OPERATOR_MINUS -> (firstOperandValue - secondOperandValue).toString()
-            OPERATOR_DIVIDE -> (firstOperandValue / secondOperandValue).toString()
-            OPERATOR_MULTIPLY -> (firstOperandValue * secondOperandValue).toString()
-            else -> ERROR_MESSAGE
+            when (operator) {
+                OPERATOR_SUM -> {
+                    result =
+                        (firstOperandUtils.getValue() + secondOperandUtils.getValue()).toString()
+                    error = ResultUtils.SUCCESS
+                }
+                OPERATOR_MINUS -> {
+                    result =
+                        (firstOperandUtils.getValue() - secondOperandUtils.getValue()).toString()
+                    error = ResultUtils.SUCCESS
+                }
+                OPERATOR_DIVIDE -> if (secondOperandUtils.getValue() == ZERO_NUMBER_DOUBLE_TYPE) {
+                    result = EMPTY_STRING
+                    error = ResultUtils.ERROR_MESSAGE_DIVISION
+                } else {
+                    result =
+                        (firstOperandUtils.getValue() / secondOperandUtils.getValue()).toString()
+                    error = ResultUtils.SUCCESS
+                }
+                OPERATOR_MULTIPLY -> {
+                    result =
+                        (firstOperandUtils.getValue() * secondOperandUtils.getValue()).toString()
+                    error = ResultUtils.SUCCESS
+                }
+                else -> {
+                    ResultUtils.ERROR_MESSAGE.toString()
+                    result = EMPTY_STRING
+                }
+            }
+            updateFirstOperand()
+            secondOperandUtils.eraseOperands()
         }
     }
 
+    override fun getResult(): String = result
+
+    private fun updateFirstOperand() {
+        if (result.substring(POSITION_ZERO).equals(OPERATOR_MINUS)) {
+            firstOperandUtils.signs = OPERATOR_MINUS
+            firstOperandUtils.value = result.substring(POSITION_ONE, result.length)
+        } else {
+            firstOperandUtils.signs = EMPTY_STRING
+            firstOperandUtils.value = result
+        }
+        operator = EMPTY_STRING
+    }
+
     override fun eraseResult(): String {
         result = EMPTY_STRING
-        firstOperand = EMPTY_STRING
-        secondOperand = EMPTY_STRING
+        firstOperandUtils.eraseOperands()
+        secondOperandUtils.eraseOperands()
         operator = EMPTY_STRING
         return EMPTY_STRING
     }
 
-    override fun saveOperationSymbol(operatorSymbol: String) {
-        operator = operatorSymbol
+    override fun saveOperationSymbol(operatorSymbol: String) = when {
+        firstOperandUtils.isEmpty() -> {
+            firstOperandUtils.signs = OPERATOR_MINUS
+        }
+        operator.isEmpty() -> {
+            operator = operatorSymbol
+        }
+        else -> secondOperandUtils.signs = OPERATOR_MINUS
     }
 
+    override fun getError() = error
```

Comment:
same here, this may be called getResult()

Reply:
Done, now its called getResultOperation (becausa i had another method named getResult)